needy/serialization.py

```diff
@@ -0,0 +1,39 @@
+import json
+
+from collections import OrderedDict
+
+
+def load_yaml(content):
+    try:
+        import yaml
+    except ImportError:
+        raise NotImplementedError('pyyaml must be installed to load YAML content')
+
+    class OrderedLoader(yaml.SafeLoader):
+        pass
+
+    def construct_mapping(loader, node):
+        loader.flatten_mapping(node)
+        return OrderedDict(loader.construct_pairs(node))
+
+    OrderedLoader.add_constructor(yaml.resolver.BaseResolver.DEFAULT_MAPPING_TAG, construct_mapping)
+
+    return yaml.load(content, OrderedLoader)
+
+
+def load_json(content):
+    return json.loads(content, object_pairs_hook=OrderedDict)
+
+
+def load_dictionary(content):
```

Comment:
I don't like this interface. If you make a JSON typo without YAML installed, it'll give you an outright incorrect error.

Manual editing of the configuration files should not be the encouraged practice. And even if you allow manual editing and YAML, how will you determine what format to write back to it with when you add commands that modify the configuration?

I think my vote is for JSON config files and a good command set to edit them.